File: stock-ai-backend/app/schemas/recommendation.py
from pydantic import BaseModel
from typing import Optional, Any, List
from datetime import datetime


class RecommendationResponse(BaseModel):
    id: int
    stock_id: str
    strategy_id: int
    signal: str
    score: Optional[float] = None
    ai_summary: Optional[str] = None
    ai_detail: Optional[str] = None
    indicators: Optional[Any] = None
    price_at: Optional[float] = None
    target_price: Optional[float] = None
    stop_loss: Optional[float] = None
    confidence: Optional[float] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class AnalysisRequest(BaseModel):
    ticker: str
    is_domestic: bool = True

File: stock-ai-frontend/src/types/recommendation.ts
import type { Indicator } from "./indicator";

export type SignalType = "BUY" | "SELL" | "HOLD" | "WATCH";
export type RiskLevel  = "LOW" | "MEDIUM" | "HIGH";

export interface Recommendation {
  id: number;
  stockId: string;
  stockName: string;
  market: string;
  strategyName: string;
  signal: SignalType;
  score: number;
  confidence: number;
  aiSummary: string;
  aiDetail?: string;
  targetPrice?: number;
  stopLoss?: number;
  priceAt?: number;
  riskLevel?: RiskLevel;
  keyPoints?: string[];
  indicators?: Partial<Indicator>;
  createdAt: string;
}

export interface AnalysisResult {
  ticker: string;
  stockName: string;
  market: string;
  currentPrice?: number;
  analysis: {
    signal: SignalType;
    confidence: number;
    summary: string;
    riskLevel: RiskLevel;
    keyPoints?: string[];
    shortTerm?: {
      outlook: string;
      buyPrice?: number;
      sellPrice?: number;
      stopLoss?: number;
      reason?: string;
    };
    midTerm?: {
      outlook: string;
      targetPrice?: number;
      keyLevel?: number;
      reason?: string;
    };
    longTerm?: {
      outlook: string;
      targetPrice?: number;
      reason?: string;
    };
    maAnalysis?: {
      arrangement: string;
      summary: string;
    };
    llmProvider?: string;
  };
  strategySignals?: Array<{
    strategy: string;
    signal: string;
    score: number;
    reason: string;
  }>;
  indicators?: Record<string, number | null>;
}
